internal/catalog/internal/controllers/workloadhealth/controller_test.go

```diff
@@ -79,18 +86,31 @@ type controllerSuite struct {
 	suite.Suite
 	client  pbresource.ResourceServiceClient
 	runtime controller.Runtime
+
+	isEnterprise bool
+	tenancy      *pbresource.Tenancy
 }
 
 func (suite *controllerSuite) SetupTest() {
 	suite.client = svctest.RunResourceService(suite.T(), types.Register)
 	suite.runtime = controller.Runtime{Client: suite.client, Logger: testutil.Logger(suite.T())}
+	suite.isEnterprise = (structs.NodeEnterpriseMetaInDefaultPartition().PartitionOrEmpty() == "default")
+
+	tenancy := resource.DefaultNamespacedTenancy()
+	if suite.isEnterprise {
+		tenancy = &pbresource.Tenancy{
+			Namespace: "foo",
+			Partition: "bar",
+		}
+	}
+	suite.tenancy = tenancy
```

Comment:
It would be nice to run the following combinations of tenancy on enterprise:
1. Non-default namespace + default partition
2. Default namespace + non-default partition
3. Non-default namespace + non-default partition
4. Default namespace + default partition

It's probably easiest to do that by setting a slice of tenancies here and running each test case with multiple different tenanceis. We do something similar to this here https://github.com/hashicorp/consul/blob/main/internal/catalog/internal/types/failover_policy_test.go#L781

Reply:
Thanks for the pointer. I did take a look at such a pattern. Should this be followed for all the test cases? Or only the main ones?

Reply:
Optimally, we would take time up front to create a pattern that makes adding tests with different tenancies trivial.

Reply:
I've updated the suite to accommodate the tenancies. Can you check the same?